lib/collection/src/shards/replica_set/snapshots.rs

```diff
@@ -91,6 +96,68 @@ impl ShardReplicaSet {
             return Ok(false);
         }
 
+        let segments_path = LocalShard::segments_path(replica_path);
+
+        let mut snapshot_segments = HashSet::new();
+        let mut snapshot_manifests = SegmentManifests::default();
+
+        for segment_entry in segments_path.read_dir()? {
+            let segment_path = segment_entry?.path();
+
+            if !segment_path.is_dir() {
+                log::warn!(
+                    "segment path {} in extracted snapshot {} is not a directory",
+                    segment_path.display(),
+                    replica_path.display(),
+                );
+
+                continue;
+            }
+
+            let segment_id = segment_path
+                .file_name()
+                .and_then(|segment_id| segment_id.to_str())
+                .expect("segment path ends with a valid segment id");
+
+            let segment_added = snapshot_segments.insert(segment_id.to_string());
+            debug_assert!(segment_added);
+
+            let manifest_path = segment_path.join("segment_manifest.json");
+
+            if !manifest_path.exists() {
+                log::debug!("segment {segment_id} does not contain segment_manifest.json");
```

Comment:
Does this deserve to be a warning?

Reply:
Regular snapshots do not contain manifests.